src/ASM/LR/ASMu2Dmxrecovery.C

```diff
@@ -68,30 +69,27 @@ bool ASMu2Dmx::assembleL2matrices (SparseMatrix& A, StdVector& B,
 
   // === Assembly loop over all elements in the patch ==========================
 
-  std::vector<LR::Element*>::iterator el1 = m_basis[geoBasis-1]->elementBegin();
-  for (int iel = 1; el1 != m_basis[geoBasis-1]->elementEnd(); ++el1, ++iel)
+  for (auto el1 : m_basis[geoBasis-1]->getAllElements())
```

Comment:
I have mixed feelings for this `auto`keyword. Although convenient for programmers, it makes the code harder to read when you have to deduce what kind of object this by checking the RHS expression. When the actual type is "simple", like here (`LR::Element*` I persume), I prefer to use the explicit type.

Reply:
i know, just sometimes i forget. changed.